Foundation/NSDate.swift

```diff
@@ -174,6 +174,28 @@ extension CFDateRef : _NSBridgable {
     internal var _nsObject: NSType { return unsafeBitCast(self, NSType.self) }
 }
 
+extension NSDate : Comparable { }
+
+public func ==(lhs: NSDate, rhs: NSDate) -> Bool {
+    return lhs === rhs || lhs.compare(rhs) == .OrderedSame
+}
+
+public func <(lhs: NSDate, rhs: NSDate) -> Bool {
+    return lhs.compare(rhs) == .OrderedAscending
+}
+
+public func <=(lhs: NSDate, rhs: NSDate) -> Bool {
+    return lhs < rhs || lhs == rhs
+}
+
+public func >(lhs: NSDate, rhs: NSDate) -> Bool {
+    return lhs.compare(rhs) == .OrderedDescending
+}
+
+public func >=(lhs: NSDate, rhs: NSDate) -> Bool {
+    return lhs > rhs || lhs == rhs
```

Comment:
Hi @tomaskraina. Great, simple and readable solution :+1: 
I'd like to suggest a small performance optimisation here: if `lhs` and `rhs` are equal you actually call the compare method 2 times on the same pair. Once inside the `>` method and then once again in `==`. Instead, you can save the result of compare and use `switch` to accept the valid cases or even use some pattern matching magic :-)
Since this is the foundation library, IMO we need to be conscious about performance decisions, even if they're small.